plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/StandardProjectWizard.java

```diff
@@ -36,45 +61,51 @@ public void addPages() {
 
   @Override
   public boolean performFinish() {
-    AnalyticsPingManager.getInstance().sendPing(
-        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_COMPLETE,
-        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE,
-        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_NATIVE);
-
-    if (config.getCloudSdkLocation() == null) {
-      File location = CloudSdkPrompter.getCloudSdkLocation(getShell());
-      if (location == null) {
-        return false;
+    if (hasAppEngineComponent) {
+      AnalyticsPingManager.getInstance().sendPing(
+          AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_COMPLETE,
+          AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE,
+          AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_NATIVE);
+
+      if (config.getCloudSdkLocation() == null) {
+        File location = CloudSdkPrompter.getCloudSdkLocation(getShell());
+        if (location == null) {
+          return false;
+        }
+        config.setCloudSdkLocation(location);
       }
-      config.setCloudSdkLocation(location);
-    }
-    // todo is this the right time/place to grab these?
-    config.setAppEngineProjectId(page.getAppEngineProjectId());
-    config.setPackageName(page.getPackageName());
 
-    config.setProject(page.getProjectHandle());
-    if (!page.useDefaults()) {
-      config.setEclipseProjectLocationUri(page.getLocationURI());
-    }
+      if ((page != null) && (page instanceof AppEngineStandardWizardPage)) {
```

Comment:
`(page != null)` is unnecessary, since `null instanceof Object` is safely false.